### PR TITLE
Deprecate `is_NumberFieldOrder`, `is_AbsoluteNumberField`, `is_RelativeNumberField`, `is_NumberFieldIdeal`, `is_NumberFieldFractionalIdeal`, `is_NumberFieldFractionalIdeal_rel`

### PR DESCRIPTION
**Pull request by mkoeppe**
### :memo: Checklist

- [x] The title is concise and informative.
- [ ] The description explains in detail what this PR is about.
- [ ] I have linked a relevant issue or discussion.
- [ ] I have created tests covering the changes.
- [ ] I have updated the documentation and checked the documentation preview.

### CODE REVIEW

File: src/sage/rings/polynomial/multi_polynomial_libsingular.pyx [hunk 1]
```diff
@@ -5665,7 +5664,7 @@ cdef class MPolynomial_libsingular(MPolynomial_libsingular_base):
             prec = 53
 
         K = FractionField(self.base_ring())
-        if K not in NumberFields() or is_NumberFieldOrder(K):
+        if K not in NumberFields() and not isinstance(K, sage.rings.abc.Order) and K != ZZ:
             raise TypeError("must be over a Numberfield or a Numberfield order")
```
Comment (kwankyu) [inline src/sage/rings/polynomial/multi_polynomial_libsingular.pyx:5669]: Original line 5668 and the modified line 5667 are not logically equivalent. Considering the error message in line 5668, it seems that the original line was wrong. Do you think so?

**Comment by kwankyu** [comment]
Otherwise lgtm.

File: src/sage/rings/polynomial/multi_polynomial_libsingular.pyx (hunk 1 above)
Comment (mkoeppe) [reply src/sage/rings/polynomial/multi_polynomial_libsingular.pyx:5669]: Yes, I think I fixed a bug there.

File: src/sage/rings/polynomial/multi_polynomial_libsingular.pyx [hunk 2]
```diff
@@ -5709,7 +5708,7 @@ cdef class MPolynomial_libsingular(MPolynomial_libsingular_base):
             1.0
         """
         K = FractionField(self.base_ring())
-        if K not in NumberFields() or is_NumberFieldOrder(K):
+        if K not in NumberFields() and not isinstance(K, sage.rings.abc.Order) and K != ZZ:
```
Comment (mkoeppe) [inline src/sage/rings/polynomial/multi_polynomial_libsingular.pyx:5712]: Also here.

File: src/sage/rings/polynomial/multi_polynomial_libsingular.pyx (hunk 1 above)
Comment (kwankyu) [reply src/sage/rings/polynomial/multi_polynomial_libsingular.pyx:5669]: It makes sense. OK.

File: src/sage/rings/polynomial/multi_polynomial_libsingular.pyx (hunk 2 above)
Comment (kwankyu) [reply src/sage/rings/polynomial/multi_polynomial_libsingular.pyx:5712]: Yes.

**Review by kwankyu — APPROVED** [review]
LGTM.

**Comment by mkoeppe** [comment]
Thanks!

**Pull request merged**